src/main/java/ga/rugal/food/springmvc/controller/MenuAction.java

```diff
@@ -42,22 +45,32 @@
 
     @Autowired
     private MenuService menuService;
+    
+    @Autowired
+    private RestaurantService restaurantService;
 
+    /**
+     * Get menu information through URL /menu
+     * This method is to get a random menu from a menu list associated with specific restaurant
+     * 
+     * @return Give successful message and menu data in JSON format if the menu exist, 
+     * on the contrary, return failed message if menu can't be found
+     */
     @ResponseBody
     @RequestMapping(method = RequestMethod.GET)
-    public Message randomMenu()
-    {
-        int total = menuService.countTotal();
-        LOG.debug(CommonLogContent.MENU_NUMBER, total);
-        if (0 == total)
-        {
+    public Message menuByRestaurant() {
+  
+        Restaurant restaurant = restaurantService.getRandomRestaurant();
```

Comment:
This method needs enhancement:  
1. Getting a random restaurant before counting number of restaurant is not logically valid.  
2. Restaurant object is not guaranteed to be valid unless db provision at least 1 count of restaurant.  
3. `getRandomRestaurant()` method and query could be avoid if there is no available restaurant.